tests/engine/test_work_chain.py

```diff
@@ -780,6 +780,182 @@ def result(self):
 
         run_and_check_success(Workchain)
 
+    def test_nested_to_context(self):
+        val = Int(5).store()
+
+        test_case = self
+
+        class SimpleWc(WorkChain):
+
+            @classmethod
+            def define(cls, spec):
+                super().define(spec)
+                spec.outline(cls.result)
+                spec.outputs.dynamic = True
+
+            def result(self):
+                self.out('result', val)
+
+        class Workchain(WorkChain):
+
+            @classmethod
+            def define(cls, spec):
+                super().define(spec)
+                spec.outline(cls.begin, cls.result)
+
+            def begin(self):
+                self.to_context(**{'sub1.sub2.result_a': self.submit(SimpleWc)})
+                return ToContext(**{'sub1.sub2.result_b': self.submit(SimpleWc)})
+
+            def result(self):
+                test_case.assertEqual(self.ctx.sub1.sub2.result_a.outputs.result, val)
+                test_case.assertEqual(self.ctx.sub1.sub2.result_b.outputs.result, val)
+
+        run_and_check_success(Workchain)
+
+    def test_nested_to_context_with_append(self):
+        val1 = Int(5).store()
+        val2 = Int(6).store()
+
+        test_case = self
+
+        class SimpleWc1(WorkChain):
+
+            @classmethod
+            def define(cls, spec):
+                super().define(spec)
+                spec.outline(cls.result)
+                spec.outputs.dynamic = True
+
+            def result(self):
+                self.out('result', val1)
+
+        class SimpleWc2(WorkChain):
+
+            @classmethod
+            def define(cls, spec):
+                super().define(spec)
+                spec.outline(cls.result)
+                spec.outputs.dynamic = True
+
+            def result(self):
+                self.out('result', val2)
+
+        class Workchain(WorkChain):
+
+            @classmethod
+            def define(cls, spec):
+                super().define(spec)
+                spec.outline(cls.begin, cls.result)
+
+            def begin(self):
+                self.to_context(**{'sub1.workchains': append_(self.submit(SimpleWc1))})
+                return ToContext(**{'sub1.workchains': append_(self.submit(SimpleWc2))})
```

Comment:
Any particular reason you are using two different subworkchain classes that do the same? Using the same `SimpleWc` work chain should achieve the same test coverage, would it not?

Reply:
I wanted to be able to distinguish by value that the order was preserved.
Don't remember why anymore, though.

Reply:
This behavior of `append_` is already tested elsewhere though and so wouldn't really be necessary here, where you are just testing the `ToContext` and `to_context` interface. Still, maybe doesn't hurt, just adds extra lines and makes it a bit more cumbersome to read, but we can keep it this way.

Reply:
ok :)